oak-segment-aws/src/main/java/org/apache/jackrabbit/oak/segment/aws/AwsContext.java

```diff
@@ -0,0 +1,441 @@
+import com.amazonaws.services.dynamodbv2.model.KeySchemaElement;
+import com.amazonaws.services.dynamodbv2.model.KeyType;
+import com.amazonaws.services.dynamodbv2.model.ProvisionedThroughput;
+import com.amazonaws.services.dynamodbv2.model.ScalarAttributeType;
+import com.amazonaws.services.dynamodbv2.util.TableUtils;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.CopyObjectRequest;
+import com.amazonaws.services.s3.model.DeleteObjectsRequest;
+import com.amazonaws.services.s3.model.DeleteObjectsRequest.KeyVersion;
+import com.amazonaws.util.TimingInfo;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.ListObjectsV2Request;
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+
+import org.apache.jackrabbit.oak.commons.Buffer;
+import org.apache.jackrabbit.oak.segment.spi.monitor.RemoteStoreMonitor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class AwsContext {
+
+    private static final Logger log = LoggerFactory.getLogger(AwsContext.class);
+
+    private static final String TABLE_ATTR_TIMESTAMP = "timestamp";
+
+    private static final String TABLE_ATTR_FILENAME = "filename";
+
+    public static final String TABLE_ATTR_CONTENT = "content";
+
+    private static final int TABLE_MAX_BATCH_WRITE_SIZE = 25;
+
+    private static final String LOCKTABLE_KEY = "key";
+
+    private final AmazonS3 s3;
+    private final String bucketName;
+    private final String rootDirectory;
+
+    private final AmazonDynamoDB ddb;
+    private final Table journalTable;
+    private final String lockTableName;
+
+    private RemoteStoreMonitor monitor;
+
+    private AwsContext(AmazonS3 s3, String bucketName, String rootDirectory, AmazonDynamoDB ddb,
+            String journalTableName, String lockTableName) {
+        this.s3 = s3;
+        this.bucketName = bucketName;
+        this.rootDirectory = rootDirectory.endsWith("/") ? rootDirectory : rootDirectory + "/";
+        this.ddb = ddb;
+        this.journalTable = new DynamoDB(ddb).getTable(journalTableName);
+        this.lockTableName = lockTableName;
+    }
+
+    private AwsContext(Configuration configuration) {
+        String region = configuration.region();
+        String rootDirectory = configuration.rootDirectory();
+        if (rootDirectory != null && rootDirectory.length() > 0 && rootDirectory.charAt(0) == '/') {
+            rootDirectory = rootDirectory.substring(1);
+        }
+
+        AWSCredentials credentials = configuration.sessionToken() == null || configuration.sessionToken().isEmpty()
```

Comment:
Per Miroslav's comments: 

General security reccomendation in AWS, when consuming services (S3 and DynamoDB in this case) is to prefer using temporary security credentials instead of long-term access keys. 
https://docs.aws.amazon.com/general/latest/gr/aws-access-keys-best-practices.html#use-roles

Seems that class `AwsContext.java` is using long-term access keys.
Link below explains how AWS IAM roles can be associated with k8n service account. 
https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html

Credentials provider that is used currently is fine during development/testing.

Reply:
It supports both. Long-term access is obtained by using accessKey and secretKey of either the root user or an IAM user.

Temporary security credentials work by using a sessionToken in conjunction with accesKey and secretKey. More details here: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#instance-metadata-security-credentials

The current implementation does support sessionToken and uses it when specified.